File: ruby/helloworld.rb
#!/usr/bin/env ruby
require "docopt"
require "pp"

doc = <<DOCOPT
Usage:
    helloworld [-n name]

Options:
    -n name   name to output [default: World]

Description:
    Basic Ruby program
DOCOPT

begin
    args = Docopt::docopt(doc)
rescue Docopt::Exit => e
    puts e.message
end

puts "Hello #{args['-n']}!"

File: python/helloworld.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Usage:
    helloworld [-n name]

Options:
    -n name     name to output [default: World]

Description:
    Basic python program
"""
from docopt import docopt


def main():
    args = docopt(__doc__)
    name = args["-n"]
    print("Hello " + name + "!")

if __name__ == "__main__":
    main()
